### PR TITLE
fix: incorrect confirmed transaction notification

### PR DESCRIPTION
**Pull request by blackdevelopa**
## **Description**
Whenever I send a transaction that fails, I can see how the popup with Transaction Confirmed message appears on the screen.

## **Manual testing steps**

_1. Step1:_ Go to test dapp
_2. Step2:_ Deploy an ERC721 token
_3. Step3:_ Mint token
_4. Step4:_ Click Transfer
_5. Step5:_ Accept
_6. Step6:_ Before tx is confirmed Click Approve
_7. Step7:_ Accept Tx
_8. Step8:_ See how Confirmed Transaction popup is displayed
_9. Step9:_ Check on Etherscan
_10. Step10:_ See how transaction failed

Bitrise: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e32dccac-1823-4b94-9f4b-7917f08a0cde

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/29962968/62c4d491-a95e-493c-95ab-9d77f2f63978

### **After**
https://recordit.co/8dMu4U47sB

## **Related issues**

_Fixes #6834 

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by blackdevelopa** [comment]
Find the relating core patch [here](https://github.com/MetaMask/core/pull/1773)

**Review by digiwand — DISMISSED** [review]

File: patches/@metamask+transaction-controller+5.0.0.patch [hunk 1]
```diff
@@ -1294,16 +1294,28 @@ index 814a899..411b310 100644
...
 -                        meta.status = TransactionStatus.confirmed;
-+                        meta.status = types_1.TransactionStatus.confirmed;
-                         this.hub.emit(`${meta.id}:confirmed`, meta);
-                         return [meta, true];
+-                        this.hub.emit(`${meta.id}:confirmed`, meta);
+-                        return [meta, true];
++                        // transactions can be added to a block and still fail, so we need to check the transaction status before emitting the confirmed event
++                        const verifyTransactionStatus = yield this.checkTxReceiptStatusIsFailed(transactionHash);
++                        if (verifyTransactionStatus) {
++                            const error = new Error('Transaction failed. Error encountered during contract execution [out of gas]');
```
Comment (digiwand) [inline patches/@metamask+transaction-controller+5.0.0.patch:1307]: is it guaranteed that if the transaction does fail, the reason is [out of gas]?

Comment (blackdevelopa) [reply patches/@metamask+transaction-controller+5.0.0.patch:1307]: I left same comment on the [core](https://github.com/MetaMask/core/pull/1773#discussion_r1346224721) repo about this.

Comment (jpuri) [reply patches/@metamask+transaction-controller+5.0.0.patch:1307]: I left some feedback in core repo: https://github.com/MetaMask/core/pull/1773#discussion_r1346982807

Comment (blackdevelopa) [reply patches/@metamask+transaction-controller+5.0.0.patch]: @jpuri Thanks for the feedback. Implemented the change [here](https://github.com/MetaMask/metamask-mobile/pull/7350/commits/1fe32f88c0f35c5e72afc72527949da10b89af6f)

Comment (blackdevelopa) [reply patches/@metamask+transaction-controller+5.0.0.patch]: @digiwand Thanks for question. Implemented the change https://github.com/MetaMask/metamask-mobile/pull/7350/commits/1fe32f88c0f35c5e72afc72527949da10b89af6f

**Review by jpuri — APPROVED** [review]

**Review by digiwand — APPROVED** [review]

**Comment by seaona** [comment]
I can see how failing transactions are displayed as `Failed`and the notification displays the `Something went wrong` message and this also works with a different locale. cc @blackdevelopa 

https://github.com/MetaMask/metamask-mobile/assets/54408225/d26efb96-d7b9-4efe-ad73-6dd2e4a239c4


https://github.com/MetaMask/metamask-mobile/assets/54408225/7fbc2248-fae2-4c2e-82f8-07b9a4c3cbb5

**Pull request merged**